app/client/src/components/shared/WaterbodyInfo.js

```diff
@@ -1148,75 +1088,152 @@ function WaterbodyInfo({
     content = congressionalDistrictContent();
   }
 
-  if (isPopup) {
-    const huc12 = clickedHuc?.data?.huc12;
-    const watershed = clickedHuc?.data?.watershed;
+  return content;
+}
 
-    content = (
-      <div css={popupContainerStyles}>
-        {clickedHuc && (
-          <>
-            {clickedHuc.status === 'no-data' && null}
-            {clickedHuc.status === 'fetching' && <LoadingSpinner />}
-            {clickedHuc.status === 'failure' && <p>Web service error</p>}
-            {clickedHuc.status === 'success' && (
-              <>
-                {type !== 'Change Location' && (
-                  <p css={popupTitleStyles}>Change to this location?</p>
-                )}
-
-                <div css={changeWatershedContainerStyles}>
-                  <div>
-                    {labelValue('WATERSHED', `${watershed} (${huc12})`)}
-                  </div>
-
-                  <div css={buttonsContainer}>
-                    <button
-                      css={buttonStyles}
-                      title="Change to this location"
-                      className="btn"
-                      onClick={(ev) => {
-                        // Clear all data before navigating.
-                        // The main reason for this is better performance
-                        // when doing a huc search by clicking on the state map. The app
-                        // will attempt to use all of the loaded state data, then clear it
-                        // then load the huc. This could take a long time if the state
-                        // has a lot of waterbodies.
-                        if (resetData) resetData();
-
-                        let baseRoute = `/community/${huc12}`;
-
-                        // community will attempt to stay on the same tab
-                        // if available, stay on the same tab otherwise go to overview
-                        let urlParts = window.location.pathname.split('/');
-                        if (
-                          urlParts.includes('community') &&
-                          urlParts.length > 3
-                        ) {
-                          window.location.assign(`${baseRoute}/${urlParts[3]}`);
-                          return;
-                        }
-
-                        window.location.assign(`${baseRoute}/overview`);
-                      }}
-                    >
-                      Yes
-                    </button>
-                  </div>
-                </div>
-              </>
-            )}
-          </>
-        )}
+type MapPopupProps = {
+  type: string,
+  feature: ?Object,
```

Comment:
Some type errors stem from this: if the `feature` attribute is nullable, then there should be null checks in the component. I don't see any calls `MapPopup` that would have a null `feature` prop, so you could probably remove the nullable indicator, but it might be safer to add the checks anyway.

Reply:
Good catch. I went removed the nullability from the `feature` prop from both `WaterbodyInfo` and `MapPopup`, since we pass that prop for every instance of `MapPopup` and `WaterbodyInfo`.